lib/galaxy/webapps/galaxy/services/visualizations.py

```diff
@@ -60,3 +97,284 @@ def index(
+
+            # Create and save first visualization revision
+            revision = VisualizationRevision(visualization=visualization, title=title, config=config, dbkey=dbkey)
+            visualization.latest_revision = revision
+
+            if save:
+                session = trans.sa_session
+                session.add(revision)
+                with transaction(session):
+                    session.commit()
+
+        return VisualizationCreateResponse(id=str(visualization.id))
+
+    def update(
+        self,
+        trans: ProvidesUserContext,
+        visualization_id: DecodedDatabaseIdField,
+        payload: VisualizationUpdatePayload,
+    ) -> Optional[VisualizationUpdateResponse]:
+        """
+        Update a visualization
+
+        :rtype:     dictionary
+        :returns:   dictionary containing Visualization details
+        """
+        rval = None
+
+        # there's a differentiation here between updating the visualization and creating a new revision
+        # that needs to be handled clearly here or alternately, using a different controller
+        # like e.g. PUT /api/visualizations/{visualization_id}/r/{revision_id}
+
+        # TODO: consider allowing direct alteration of revisions title (without a new revision)
+        #   only create a new revsion on a different config
+
+        # only update owned visualizations
+        visualization = self._get_visualization(trans, visualization_id, check_ownership=True)
+        latest_revision = cast(VisualizationRevision, visualization.latest_revision)
+        title = payload.title or latest_revision.title
+        dbkey = payload.dbkey or latest_revision.dbkey
+        deleted = payload.deleted or visualization.deleted
+        config = payload.config or latest_revision.config
+
+        latest_config = latest_revision.config
+        if (
+            (title != latest_revision.title)
+            or (dbkey != latest_revision.dbkey)
+            or (json.dumps(config) != json.dumps(latest_config))
+        ):
+            revision = self._add_visualization_revision(trans, visualization, config, title, dbkey)
+            rval = {"id": str(visualization_id), "revision": str(revision.id)}
+
+        # allow updating vis title
+        visualization.title = title
+        visualization.deleted = deleted
+        with transaction(trans.sa_session):
+            trans.sa_session.commit()
+
+        return VisualizationUpdateResponse(**rval) if rval else None
+
+    def _get_visualization(
+        self,
+        trans: ProvidesUserContext,
+        visualization_id: DecodedDatabaseIdField,
+        check_ownership=True,
+        check_accessible=False,
+    ) -> Visualization:
+        """
+        Get a Visualization from the database by id, verifying ownership.
+        """
+        try:
+            visualization = trans.sa_session.get(Visualization, visualization_id)
+        except TypeError:
+            visualization = None
```

Comment:
There's never going to be a type error, the try/except doesn't do anything

Reply:
So should I just remove the try/except?

Reply:
yes